markdown-toc.el

```diff
@@ -222,18 +229,37 @@ If called interactively with prefix arg REPLACE-TOC-P, replaces previous TOC."
   (save-excursion
     (when (markdown-toc--toc-already-present-p)
       ;; when toc already present, remove it
-      (let ((region-start (markdown-toc--toc-start))
-            (region-end   (markdown-toc--toc-end)))
-        (delete-region region-start (1+ region-end))
-        (when replace-toc-p
-          (goto-char region-start))))
+      (markdown-toc--delete-toc))
     (->> (markdown-imenu-create-nested-index)
          markdown-toc--compute-toc-structure
          (funcall markdown-toc-user-toc-structure-manipulation-fn)
          markdown-toc--generate-toc
          insert)))
 
+;;;###autoload
+(defun markdown-toc-generate-or-refresh-toc ()
+  "Generate a TOC for markdown file at current point or refreshes an already generated TOC."
+  (interactive)
+  (markdown-toc-generate-toc t))
+
+;;;###autoload
+(defun markdown-toc-refresh-toc ()
+  "Refreshes an already generated TOC."
+  (interactive)
+  (when (markdown-toc--toc-already-present-p)
+    (markdown-toc-generate-toc t)))
```

Comment:
I don't get the nuance between the 2 previous definitions, can you please clarify?